Version 3.15.1 fails to build (linking problem)

## Version of syslog-ng

3.15.1

## Platform

TLD Linux 2018.01

# Issue

Version 3.15.1 fails during build when linking syslog-ng. Problem was introduced by this change: https://github.com/balabit/syslog-ng/commit/3ba8b3e5279aa6daa21da4bd32fd92664623a450

Here is sample output after issuing make:

x86_64-tld-linux-gcc -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros  -O2 -g -pthread -O2 -fwrapv -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fPIC -march=x86-64   -Wl,-z,relro -Wl,-z,combreloc  -o syslog-ng/syslog-ng syslog-ng/main.o -L./lib/.libs -lsyslog-ng   -lrt  -Wl,-Bstatic -Wl,--whole-archive -Wl,--export-dynamic -lgmodule-2.0 -pthread -lgthread-2.0 -pthread -lglib-2.0  ./lib/eventlog/src/.libs/libevtlog.la -lpcre   -Wl,--no-whole-archive -Wl,-Bdynamic -Wl,--whole-archive -L./lib/ivykis/src -livykis -Wl,--no-whole-archive -lcap -ldl
./lib/eventlog/src/.libs/libevtlog.la: file not recognized: File format not recognized
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:10907: syslog-ng/syslog-ng] Error 1
make[1]: *** [Makefile:17783: all-recursive] Error 1
make: *** [Makefile:7669: all] Error 2

This can be fixed in three ways:

1. Adding "libtool --mode=link" at beginning of above command and executing it manually
2. Removing line "syslog_ng_syslog_ng_LINK =  @SYSLOGNG_LINK@" from syslog-ng/Makefile.am
3. Reverting commit mentioned above

--- a/configure.ac
+++ b/configure.ac
@@ -794,7 +794,7 @@ dnl libevtlog headers/libraries (remove after relicensing libevtlog)
 dnl ***************************************************************************
 
 EVTLOG_LIBS="\$(top_builddir)/lib/eventlog/src/libevtlog.la"
-EVTLOG_NO_LIBTOOL_LIBS="\$(top_builddir)/lib/eventlog/src/.libs/libevtlog.la"
+EVTLOG_NO_LIBTOOL_LIBS="\$(top_builddir)/lib/eventlog/src/.libs/libevtlog.so"
 EVTLOG_CFLAGS="-I\$(top_srcdir)/lib/eventlog/src -I\$(top_builddir)/lib/eventlog/src"
 
 dnl ***************************************************************************
@@ -1688,7 +1688,7 @@ if test "x$linking_mode" = "xdynamic"; then
 	# syslog-ng binary is linked with the default link command (e.g. libtool)
 	SYSLOGNG_LINK='$(LINK)'
 else
-	SYSLOGNG_DEPS_LIBS="$LIBS $BASE_LIBS $RESOLV_LIBS $LD_START_STATIC -Wl,${WHOLE_ARCHIVE_OPT} $GLIB_LIBS $EVTLOG_NO_LIBTOOL_LIBS $PCRE_LIBS $REGEX_LIBS  -Wl,${NO_WHOLE_ARCHIVE_OPT} $IVYKIS_NO_LIBTOOL_LIBS $LD_END_STATIC $LIBCAP_LIBS $DL_LIBS "
+	SYSLOGNG_DEPS_LIBS="$LIBS $BASE_LIBS $RESOLV_LIBS $EVTLOG_NO_LIBTOOL_LIBS $LD_START_STATIC -Wl,${WHOLE_ARCHIVE_OPT} $GLIB_LIBS $PCRE_LIBS $REGEX_LIBS  -Wl,${NO_WHOLE_ARCHIVE_OPT} $IVYKIS_NO_LIBTOOL_LIBS $LD_END_STATIC $LIBCAP_LIBS $DL_LIBS"
 	TOOL_DEPS_LIBS="$LIBS $BASE_LIBS $GLIB_LIBS $EVTLOG_LIBS $RESOLV_LIBS $LIBCAP_LIBS $PCRE_LIBS $REGEX_LIBS $IVYKIS_LIBS $DL_LIBS"
 	CORE_DEPS_LIBS=""
 

--- a/modules/diskq/Makefile.am
+++ b/modules/diskq/Makefile.am
@@ -46,8 +46,8 @@ modules_diskq_libdisk_buffer_la_DEPENDENCIES = $(MODULE_DEPS_LIBS) $(LIBSYSLOG_N
 
 modules_diskq_dqtool_SOURCES = modules/diskq/dqtool.c
 modules_diskq_dqtool_LDADD = \
-  $(TOOL_DEPS_LIBS) \
   $(MODULE_DEPS_LIBS) \
+  $(TOOL_DEPS_LIBS) \
   $(LIBSYSLOG_NG_DISK_BUFFER)
 
 modules/diskq modules/diskq/ mod-diskq: modules/diskq/libdisk-buffer.la \

--- a/syslog-ng-ctl/Makefile.am
+++ b/syslog-ng-ctl/Makefile.am
@@ -10,5 +10,7 @@ syslog_ng_ctl_syslog_ng_ctl_SOURCES		= 	\
 EXTRA_DIST					+=	\
 	syslog-ng-ctl/control-client-unix.c
 
-syslog_ng_ctl_syslog_ng_ctl_LDADD		= lib/libsyslog-ng.la @BASE_LIBS@ @GLIB_LIBS@ @RESOLV_LIBS@
-
+syslog_ng_ctl_syslog_ng_ctl_LDADD		= \
+	$(MODULE_DEPS_LIBS) \
+	$(TOOL_DEPS_LIBS)
+syslog_ng_ctl_syslog_ng_ctl_DEPENDENCIES	= lib/libsyslog-ng.la

